File: ployglots/mta-ployglot-pkg-figma-api/packages/ts/src/resources/comments.ts
// @ts-nocheck
import type { FigmaClient } from '../figma-client.js';
import type { FigmaComment, FigmaCommentsResponse } from './types.js';

export class CommentsResource {
  constructor(private readonly client: FigmaClient) {}

  /** GET /v1/files/:key/comments — all comments on a file. */
  async list(fileKey: string): Promise<FigmaComment[]> {
    const resp = await this.client.get<FigmaCommentsResponse>(
      `/v1/files/${encodeURIComponent(fileKey)}/comments`,
    );
    return resp.comments ?? [];
  }

  /** POST /v1/files/:key/comments — create a comment. */
  async create(
    fileKey: string,
    body: { message: string; client_meta?: unknown; comment_id?: string },
  ): Promise<FigmaComment> {
    return this.client.post<FigmaComment>(`/v1/files/${encodeURIComponent(fileKey)}/comments`, {
      body,
    });
  }

  /** DELETE /v1/files/:key/comments/:id — delete a comment. */
  async delete(fileKey: string, commentId: string): Promise<void> {
    await this.client.delete<void>(
      `/v1/files/${encodeURIComponent(fileKey)}/comments/${encodeURIComponent(commentId)}`,
    );
  }

  /** GET /v1/files/:key/comments/:id/reactions — list reactions on a comment. */
  async reactions(fileKey: string, commentId: string): Promise<unknown> {
    return this.client.get<unknown>(
      `/v1/files/${encodeURIComponent(fileKey)}/comments/${encodeURIComponent(commentId)}/reactions`,
    );
  }

  /** POST /v1/files/:key/comments/:id/reactions — add a reaction. */
  async addReaction(fileKey: string, commentId: string, emoji: string): Promise<unknown> {
    return this.client.post<unknown>(
      `/v1/files/${encodeURIComponent(fileKey)}/comments/${encodeURIComponent(commentId)}/reactions`,
      { body: { emoji } },
    );
  }

  /** DELETE /v1/files/:key/comments/:id/reactions — remove a reaction. */
  async removeReaction(fileKey: string, commentId: string, emoji: string): Promise<void> {
    await this.client.delete<void>(
      `/v1/files/${encodeURIComponent(fileKey)}/comments/${encodeURIComponent(commentId)}/reactions`,
      { params: { emoji } },
    );
  }
}

File: ployglots/mta-ployglot-pkg-figma-api/packages/py/figma_api/resources/comments.py
from __future__ import annotations

from typing import TYPE_CHECKING, Any
from urllib.parse import quote

if TYPE_CHECKING:
    from .._figma_client import FigmaClient


class CommentsResource:
    def __init__(self, client: FigmaClient) -> None:
        self._client = client

    async def list(self, file_key: str) -> list[dict[str, Any]]:
        """GET /v1/files/:key/comments — all comments on a file."""
        resp = await self._client.get(f"/v1/files/{quote(file_key, safe='')}/comments")
        return resp.get("comments", []) if isinstance(resp, dict) else []

    async def create(
        self,
        file_key: str,
        *,
        message: str,
        client_meta: Any = None,
        comment_id: str | None = None,
    ) -> dict[str, Any]:
        """POST /v1/files/:key/comments — create a comment."""
        body: dict[str, Any] = {"message": message}
        if client_meta is not None:
            body["client_meta"] = client_meta
        if comment_id is not None:
            body["comment_id"] = comment_id
        return await self._client.post(
            f"/v1/files/{quote(file_key, safe='')}/comments",
            json=body,
        )

    async def delete(self, file_key: str, comment_id: str) -> None:
        """DELETE /v1/files/:key/comments/:id — delete a comment."""
        await self._client.delete(
            f"/v1/files/{quote(file_key, safe='')}/comments/{quote(comment_id, safe='')}"
        )

    async def reactions(self, file_key: str, comment_id: str) -> dict[str, Any]:
        """GET /v1/files/:key/comments/:id/reactions — list reactions on a comment."""
        return await self._client.get(
            f"/v1/files/{quote(file_key, safe='')}/comments/"
            f"{quote(comment_id, safe='')}/reactions"
        )

    async def add_reaction(
        self, file_key: str, comment_id: str, emoji: str
    ) -> dict[str, Any]:
        """POST /v1/files/:key/comments/:id/reactions — add a reaction."""
        return await self._client.post(
            f"/v1/files/{quote(file_key, safe='')}/comments/"
            f"{quote(comment_id, safe='')}/reactions",
            json={"emoji": emoji},
        )

    async def remove_reaction(self, file_key: str, comment_id: str, emoji: str) -> None:
        """DELETE /v1/files/:key/comments/:id/reactions — remove a reaction."""
        await self._client.delete(
            f"/v1/files/{quote(file_key, safe='')}/comments/"
            f"{quote(comment_id, safe='')}/reactions",
            params={"emoji": emoji},
        )
